plugins/scaffolder-backend/src/service/router.ts

```diff
@@ -445,7 +490,21 @@ export async function createRouter(
         });
       },
     )
-    .get('/v2/actions', async (_req, res) => {
+    .get('/v2/actions', async (req, res) => {
+      const credentials = await httpAuth.credentials(req);
+
+      if (permissions) {
```

Comment:
These blocks to check the permissions are repeated many times in each API.
To lessen the boilerplate, some common function something like `checkPermissions` can be created.

Reply:
Updated in https://github.com/backstage/backstage/pull/24518/commits/83643ef0919f5dadcdb4ea7ea62bcf593dc04698